### DISCUSSION TITLE
typescript.nuxtjs.org class api is actual for nuxt3?

### DISCUSSION THREAD

**Discussion by n4an** [nuxt/nuxt#17745] [General]
1 typescript.nuxtjs.org class api is actual for nuxt3? i rewrite components to class api when nuxt 3 doesn't be released.

2 with nuxt 3 server how to catch routes with dinamic params (/api/test/:id)

**Comment by Diizzayy** [comment]
@n4an 

> typescript.nuxtjs.org class api is actual for nuxt3?

No, that documentation is for using typescript with nuxt 2 and is no longer recommended in any regard.

> with nuxt 3 server how to catch routes with dinamic params 

Assuming that you indeed are referring to using dynamic params with [api routes](https://v3.nuxtjs.org/guide/directory-structure/server#api-routes)

`/server/api/`
```ts
import { createApp, createRouter } from 'h3'

const app = createApp()
const router = createRouter()

app.use(router)

router.get("/:id", (req, res) => {
  const id = req.params?.id
  return 'Hello: ' + id
})

export default app
```

**Reply by n4an** [reply]
i think that it can be done by naming the file in some mode (like pages)

**Reply by n4an** [reply]
for each file i need to create new app and router
```
const app = createApp()
const router = createRouter()
```

**Reply by n4an** [reply]
that `Nuxt will automatically read in any files in the ~/server/api` is only for simply routes with queries params. for other need to create app and router

Nuxt 2 was beautiful but now i think authors not understand what they do

**Reply by n4an** [reply]
that h3 need to be like nestjs but not as is. after nest or adonis is no pleasure to return back to that

**Reply by n4an** [reply]
and class api was good and composition api is sheet

**Comment by n4an** [comment]
@Diizzayy But like in next?

```
For example, the API route pages/api/post/[pid].js has the following code:

export default function handler(req, res) {
  const { pid } = req.query
  res.end(`Post: ${pid}`)
}
```

**Reply by Diizzayy** [reply]
@n4an The latest release as of today ( #3956 ) improves upon this functionality.

`/server/api/test/[id].ts`
```ts 
export default eventHandler(async (event) => {
    const params = event.context.params;
    // const body = await useBody(event.req);  // only for POST | PUT | PATCH | DELETE requests
    const cookies = useCookies(event.req);

    return `Hello, ${params.id}!`
})
```
